File: 07-stacks-queues/09_balanced_brackets.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Balanced Brackets (HackerRank) -- Medium
# Like valid parentheses but specifically the HackerRank version.
#
#   Key: Same stack matching pattern. Handle edge cases: empty string, single bracket.

def is_balanced(s):
    pass


run_tests(is_balanced, [
    (("{[()]}",),  True),
    (("{[(])}",),  False),
    (("{{[[(())]]}}", ), True),
    (("",),        True),
    (("{",),       False),
])

File: 07-stacks-queues/09_balanced_brackets.c
#include "../test_runner.h"

bool is_balanced(char* s) {
    return false;
}

int main() {
    TEST_BOOL("valid",    is_balanced("{[()]}"),       true);
    TEST_BOOL("invalid",  is_balanced("{[(])}"),       false);
    TEST_BOOL("nested",   is_balanced("{{[[(())]]}}"), true);
    TEST_BOOL("empty",    is_balanced(""),             true);
    TEST_BOOL("single",   is_balanced("{"),            false);
    TEST_SUMMARY();
}
